torch/csrc/jit/serialization/export.cpp

```diff
@@ -1366,10 +1366,22 @@ void check_onnx_proto(const std::string& proto_string, bool full_check) {
     throw std::runtime_error("Invalid ONNX proto string.");
     return;
   }
+  // baseline check
   onnx::checker::check_model(model);
-
+  onnx::shape_inference::InferShapes(model);
+  // full check including strict shape inference check
+  // Notice that onnx::checker::check_model doesn't
+  // include non-strict model onnx::shape_inference::InferShapes(model).
+  // That's why we need to write another onnx::checker::check_model(model,
+  // full_check) here.
   if (full_check) {
-    onnx::shape_inference::InferShapes(model);
+    try {
+      onnx::checker::check_model(model, full_check);
+    } catch (const onnx::InferenceError& ex) {
+      TORCH_WARN(
+          "This is an invalid ONNX model with failed ONNX shape type inference. It is not executable by ONNXRUNTIME platform. More detail: ",
```

Comment:
```suggestion
          "The exported ONNX model failed ONNX shape inference. The model will not be executable by the ONNX Runtime. If this is unintended and you believe there is a bug, please report an issue at https://github.com/pytorch/pytorch/issues. Error reported by the ONNX checker: ",
```

suggested wording (be sure to break the lines if too long)

Reply:
Done